### ISSUE TITLE
Micronaut POST request with dual protocol and http2 enabled on HTTP port

### ISSUE THREAD

**Issue by Patanouk** [micronaut-projects/micronaut-core#5629]
### Task List

- [x] Steps to reproduce provided
- [X] Stacktrace (if present) provided
- [X] Example that reproduces the problem uploaded to Github
- [X] Full description of the issue provided (see below)

### Steps to Reproduce

1. Run the application in the repository linked below (https://github.com/Patanouk/Micronaut-dualprotocol)
2. Run `curl -H "Content-type:application/json" -X POST http://localhost:8080/hello -d "hello"`
3. The curl call is hanging

It happens when both dual protocol and http2 are enabled
Here is the minimal `application.yml`, taken from the linked repository

```
micronaut:
  application:
    name: micronautDualprotocol
  server:
    port: 8080
    http-version: 2.0
    dual-protocol: true
  ssl:
    enabled: true
    port: 8443
    build-self-signed: true
```

### Expected Behaviour

POST call to HTTP port succeeds with a 200 using HTTP1.1

### Actual Behaviour

The POST call hangs on HTTP port

### Example Application

- https://github.com/Patanouk/Micronaut-dualprotocol

**Comment by graemerocher** [comment]
@yawkat related Netty issue

**Comment by yawkat** [comment]
probably fixed by #6300 , will check.

**Comment by yawkat** [comment]
Yes, tested with 3.1.3, seems to be fixed.
